internal/db/repositories/repository.go

```diff
@@ -0,0 +1,12 @@
+package repositories
+
+import (
+	"context"
+)
+
+type Repository interface {
+	Find(ctx context.Context, filter interface{}) ([]UserModel, error)
+	Create(ctx context.Context, data UserModel) (UserModel, error)
+	Delete(ctx context.Context, filter interface{}) error
+	Update(ctx context.Context, filter interface{}, update interface{}) error
```

Comment:
for filters, should we create a standard struct which will tell us what all filters are available?